easybuild/easyblocks/a/adf.py

```diff
@@ -0,0 +1,95 @@
+# the Hercules foundation (http://www.herculesstichting.be/in_English)
+# and the Department of Economy, Science and Innovation (EWI) (http://www.ewi-vlaanderen.be/en).
+#
+# http://github.com/hpcugent/easybuild
+#
+# EasyBuild is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation v2.
+#
+# EasyBuild is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with EasyBuild.  If not, see <http://www.gnu.org/licenses/>.
+##
+"""
+EasyBuild support for building and installing ADF, implemented as an easyblock
+"""
+import os
+import shutil
+
+import easybuild.tools.environment as env
+from easybuild.framework.easyblock import EasyBlock
+from easybuild.tools.build_log import EasyBuildError
+from easybuild.tools.run import run_cmd
+
+
+class EB_ADF(EasyBlock):
+    """Support for building/installing ADF."""
+
+    def __init__(self, *args, **kwargs):
+        """Initialisation of custom class variables for ADF."""
+        super(EB_ADF, self).__init__(*args, **kwargs)
+
+        self.build_in_installdir = True
+
+    def extract_step(self):
+        """Extract sources."""
+        # strip off 'adf<version>' part to avoid having everything in a subdirectory
+        self.cfg['unpack_options'] = "--strip-components=1"
+        super(EB_ADF, self).extract_step()
+
+    def configure_step(self):
+        """Custom configuration procedure for ADF."""
+
+        env.setvar('ADFHOME', self.installdir)
+        env.setvar('ADFBIN', os.path.join(self.installdir, 'bin'))
+        env.setvar('ADFRESOURCES', os.path.join(self.installdir, 'atomicdata'))
+        if self.cfg['license_file'] and os.path.exists(self.cfg['license_file']):
+            env.setvar('SCMLICENSE', self.cfg['license_file'])
+        else:
+            raise EasyBuildError("No or non-existing license file specified: %s", self.cfg['license_file'])
+
+        cmd = './Install/configure'
+        run_cmd(cmd, log_all=True, simple=True, log_ok=True)
+
+    def build_step(self):
+        """No separate custom build procedure for ADF, see install step."""
+        pass
+
+    def install_step(self):
+        """Custom install procedure for ADF."""
+
+        # this shouldn't be needed, is init.sh shipped in the wrong place?!
+        shutil.copy2(os.path.join('Install', 'init.sh'), os.path.join('bin', 'init.sh'))
+
+        cmd = "./bin/foray -j %d" % self.cfg['parallel']
+        run_cmd(cmd, log_all=True, simple=True, log_ok=True)
+
+    def sanity_check_step(self):
+        """Custom sanity check for ADF."""
+        custom_paths = {
+            'files': ['bin/adf'],
+            'dirs': ['atomicdata', 'examples'],
```

Comment:
fixed in easyconfig, both sources and binary version are available to recompile from source >_<